Add a .git-blame-ignore-revs file

reneme opened the pull request:
This cleans up `git blame` (also `tig blame` or VSCode's GitLens annotations) by automatically skipping commits that just reformatted code using clang-format. The name of the ignore-file isn't standard, but it seems to be the convention across the community.

~~Leaving this as a "Draft" for now. I personally [just learnt about this feature](https://www.michaelheap.com/git-ignore-rev/), and find it super useful (worthy of a new top-level file 😏)~~

To use this, you'll have to adapt your git config like so:

```bash
git config --local blame.ignoreRevsFile src/configs/git-blame-ignore-revs
git config --local blame.markIgnoredLines true
```

The first config enables the usage of the ignore file by default, the second one marks blame-lines with a '?' if they skipped one or more commits.

randombit reviewed (changes requested):
Useful!

Given that git doesn't understand this by default, I disagree with your assessment re top level directory. Fine for `src/configs`.

Maybe also worth a mention in the contributing guide.

reneme commented:
> Given that git doesn't understand this by default, I disagree with your assessment re top level directory. Fine for `src/configs`.

In fact, its probably better to not configure this globally after all. It seems that (at least) `tig blame` is tripped up if other repositories don't provide this file but it is configured globally in `~/.gitconfig`. Moved to `src/configs` and updated the documentation accordingly.

> Maybe also worth a mention in the contributing guide.

Done.
